src/Type/Php/PropertyExistsTypeSpecifyingExtension.php

```diff
@@ -65,34 +66,45 @@ public function specifyTypes(
 			);
 		}
 
-		if ($propertyNameType->getValue() === '') {
-			return new SpecifiedTypes([], []);
+		$hasPropertyTypes = [];
+		foreach ($propertyNameTypes as $propertyNameType) {
+			if($propertyNameType->getValue() === '') {
+				return new SpecifiedTypes([], []);
+			}
+
+			$hasPropertyTypes[] = new HasPropertyType($propertyNameType->getValue());
 		}
 
 		$objectType = $scope->getType($node->getArgs()[0]->value);
 		if ($objectType instanceof ConstantStringType) {
 			return new SpecifiedTypes([], []);
 		} elseif ($objectType->isObject()->yes()) {
-			$propertyNode = new PropertyFetch(
-				$node->getArgs()[0]->value,
-				new Identifier($propertyNameType->getValue()),
-			);
+			$propertyNodes = [];
+
+			foreach($propertyNameTypes as $propertyNameType) {
+				$propertyNodes[] = new PropertyFetch(
+					$node->getArgs()[0]->value,
+					new Identifier($propertyNameType->getValue()),
```

Comment:
identifier expecs a non-emtpy-string and within your loop you did not make sure `$propertyNameType->getValue()` is always non-empty.

you can fix this in the same way you already did in a similar loop above.